diff --git a/CMakeLists.txt b/CMakeLists.txt
index df09b8c49..740f1e946 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -454,6 +454,9 @@ target_sources(test_glift1 PRIVATE $<TARGET_OBJECTS:moduleVersion>)
 
 include (${CMAKE_CURRENT_SOURCE_DIR}/modelTests.cmake)
 
+set_tests_properties(tasklets_failure PROPERTIES DISABLED True)
+set_tests_properties(SatfuncConsistencyChecks_parallel PROPERTIES SKIP_RETURN_CODE 200)
+
 if (HAVE_OPM_TESTS)
   include (${CMAKE_CURRENT_SOURCE_DIR}/compareECLFiles.cmake)
 
diff --git a/parallelUnitTests.cmake b/parallelUnitTests.cmake
index 4036bf7a1..b7630cdf5 100644
--- a/parallelUnitTests.cmake
+++ b/parallelUnitTests.cmake
@@ -126,6 +126,7 @@ foreach(NPROC 2 3 4)
     PROCESSORS
       ${NPROC}
   )
+  set_tests_properties(parallel_satfunc_consistency_checks_np${NPROC} PROPERTIES SKIP_RETURN_CODE 200)
 endforeach()
 
 opm_add_test(test_broadcast
