diff --git a/CMakeLists.txt b/CMakeLists.txt
index b2e015b12..bdd0c6246 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -504,6 +504,9 @@ endif()
 
 include (${CMAKE_CURRENT_SOURCE_DIR}/modelTests.cmake)
 
+set_tests_properties(tasklets_failure PROPERTIES DISABLED True)
+set_tests_properties(SatfuncConsistencyChecks_parallel rftcontainer PROPERTIES SKIP_RETURN_CODE 200)
+
 if (HAVE_OPM_TESTS)
   include (${CMAKE_CURRENT_SOURCE_DIR}/compareECLFiles.cmake)
 
diff --git a/parallelUnitTests.cmake b/parallelUnitTests.cmake
index b70891de6..e4e113f04 100644
--- a/parallelUnitTests.cmake
+++ b/parallelUnitTests.cmake
@@ -109,6 +109,7 @@ foreach(NPROC 2 3 4)
     PROCESSORS
       ${NPROC}
   )
+  set_tests_properties(rftcontainer_np${NPROC} PROPERTIES SKIP_RETURN_CODE 200)
 endforeach()
 
 foreach(NPROC 2 3 4)
@@ -141,6 +142,7 @@ foreach(NPROC 2 3 4)
     PROCESSORS
       ${NPROC}
   )
+  set_tests_properties(parallel_satfunc_consistency_checks_np${NPROC} PROPERTIES SKIP_RETURN_CODE 200)
 endforeach()
 
 opm_add_test(test_broadcast
