diff --git a/cmake/scripts/extract_cmake_data.py b/cmake/scripts/extract_cmake_data.py
index d24af09a8..70acc2d30 100755
--- a/cmake/scripts/extract_cmake_data.py
+++ b/cmake/scripts/extract_cmake_data.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-FileCopyrightInfo: Copyright © DUNE Project contributors, see file LICENSE.md in module root
 # SPDX-License-Identifier: LicenseRef-GPL-2.0-only-with-DUNE-exception
 
diff --git a/cmake/scripts/run-in-dune-env.sh.in b/cmake/scripts/run-in-dune-env.sh.in
index 6a4e423d1..32ed36e06 100755
--- a/cmake/scripts/run-in-dune-env.sh.in
+++ b/cmake/scripts/run-in-dune-env.sh.in
@@ -1,4 +1,4 @@
-#!@BASH@
+#!/usr/bin/env bash
 # SPDX-FileCopyrightInfo: Copyright © DUNE Project contributors, see file LICENSE.md in module root
 # SPDX-License-Identifier: LicenseRef-GPL-2.0-only-with-DUNE-exception
 
diff --git a/cmake/scripts/set-dune-pythonpath.sh.in b/cmake/scripts/set-dune-pythonpath.sh.in
index 4fbfa24a0..687373cdc 100755
--- a/cmake/scripts/set-dune-pythonpath.sh.in
+++ b/cmake/scripts/set-dune-pythonpath.sh.in
@@ -1,4 +1,4 @@
-#!@BASH@
+#!/usr/bin/env bash
 # SPDX-FileCopyrightInfo: Copyright © DUNE Project contributors, see file LICENSE.md in module root
 # SPDX-License-Identifier: LicenseRef-GPL-2.0-only-with-DUNE-exception
 
