diff --git a/cmake/scripts/extract_cmake_data.py b/cmake/scripts/extract_cmake_data.py
index f9ba333..f04cfd4 100755
--- a/cmake/scripts/extract_cmake_data.py
+++ b/cmake/scripts/extract_cmake_data.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # SPDX-FileCopyrightInfo: Copyright (C) DUNE Project contributors, see file LICENSE.md in module root
 # SPDX-License-Identifier: LicenseRef-GPL-2.0-only-with-DUNE-exception
 
diff --git a/cmake/scripts/run-in-dune-env.sh.in b/cmake/scripts/run-in-dune-env.sh.in
index c8a52e8..52aef19 100755
--- a/cmake/scripts/run-in-dune-env.sh.in
+++ b/cmake/scripts/run-in-dune-env.sh.in
@@ -1,4 +1,4 @@
-#!@BASH@
+#!/bin/bash
 # SPDX-FileCopyrightInfo: Copyright (C) DUNE Project contributors, see file LICENSE.md in module root
 # SPDX-License-Identifier: LicenseRef-GPL-2.0-only-with-DUNE-exception
 
